### ISSUE TITLE
Missing commonly used parameters in AdaptiveLogSoftmaxWithLoss

### ISSUE THREAD

**Issue by thesofakillers** [pytorch/pytorch#89053]
### 🚀 The feature, motivation and pitch

Similarly to CrossEntropyLoss and other more common loss modules and functions, it would be nice to provide the `ignore_index` parameter in [AdaptiveLogSoftmaxWithLoss](https://pytorch.org/docs/stable/generated/torch.nn.AdaptiveLogSoftmaxWithLoss.html). The same applies to the `reduction` parameter. This would help in unifying the loss experience across pytorch. For convenience, here is the docs for the functionality of `ignore_index` in CrossEntropyLoss:

> Specifies a target value that is ignored and does not contribute to the input gradient. When size_average is True, the loss is averaged over non-ignored targets. Note that ignore_index is only applicable when the target contains class indices.

and for `reduction`:

> Specifies the reduction to apply to the output: 'none' | 'mean' | 'sum'. 'none': no reduction will be applied, 'mean': the weighted mean of the output is taken, 'sum': the output will be summed. Note: size_average and reduce are in the process of being deprecated, and in the meantime, specifying either of those two args will override reduction. Default: 'mean'

### Alternatives

Another alternative could be mathematical notation for how to implement `ignore_index` manually, similarly to how this is described in the [CrossEntropyLoss docs](https://pytorch.org/docs/stable/generated/torch.nn.CrossEntropyLoss.html#torch.nn.CrossEntropyLoss). I suppose this is related to #14318.

Further documentation on what the current reduction scheme is would be useful, since we have no control over this. I hope it is 'none', but given that a scalar is returned, I assume it's `mean`.

**Comment by thesofakillers** [comment]
closing as this is a duplicate of #12675
